src/core/lib/surface/call.cc

```diff
@@ -3609,102 +3193,78 @@ class ServerCallSpine final : public PipeBasedCallSpine,
   grpc_call_error StartBatch(const grpc_op* ops, size_t nops, void* notify_tag,
                              bool is_notify_tag_closure) override;
 
-  bool Completed() final { Crash("unimplemented"); }
-  bool failed_before_recv_message() const final { Crash("unimplemented"); }
+  Arena* arena() override { return call_handler_.arena(); }
 
-  ServerCallContext* server_call_context() override { return this; }
-  const void* server_stream_data() override { Crash("unimplemented"); }
-  void PublishInitialMetadata(
-      ClientMetadataHandle metadata,
-      grpc_metadata_array* publish_initial_metadata) override;
-  ArenaPromise<ServerMetadataHandle> MakeTopOfServerCallPromise(
-      CallArgs, grpc_completion_queue*,
-      absl::FunctionRef<void(grpc_call* call)>) override {
-    Crash("unimplemented");
+  grpc_event_engine::experimental::EventEngine* event_engine() const override {
+    return call_handler_.event_engine();
   }
 
-  void V2HackToStartCallWithoutACallFilterStack() override {}
+  void ExternalRef() override { Ref().release(); }
+  void ExternalUnref() override { Unref(); }
+  void InternalRef(const char*) override { WeakRef().release(); }
+  void InternalUnref(const char*) override { WeakUnref(); }
 
-  ClientMetadata& UnprocessedClientInitialMetadata() override {
-    Crash("not for v2");
+  void Orphaned() override {
+    // TODO(ctiller): only when we're not already finished
```

Comment:
What's this about?

Reply:
we're cancelling the call if it's early orphaned (provides signal to all other layers to wind down)
-- but if the call is already completed we shouldn't need to, so there's an optimization available here that we should take advantage of (but probably not important immediately)